packages/flutter/lib/src/cupertino/search_field.dart

```diff
@@ -0,0 +1,176 @@
+  ///
+  /// Creates a [CupertinoTextField] that mimicks the look and behavior of UIKit's
+  /// [UISearchTextField].
+  ///
+  /// Similar to [CupertinoTextField], to provide a prefilled text entry, pass
+  /// in a [TextEditingController] with an initial value to the [controller]
+  /// parameter. The [controller] property must not be null.
+  ///
+  /// The [onChanged] parameter takes a [Function(String)] which is invoked
+  /// upon a change in the text field's value.
+  ///
+  /// The [onSubmitted] parameter takes a [Function(String)] which is invoked
+  /// when the keyboard submits.
+  ///
+  /// To provide a hint placeholder text that appears when the text entry is
+  /// empty, pass a [String] to the [placeholder] parameter. This defaults to
+  /// 'Search'.
+  ///
+  /// To set the placeholder color and style, pass a [TextStyle] to the
+  /// [placeholderStyle] parameter. This defaults to the default gray iOS color.
+  ///
+  /// To set the text field's background color and border radius, pass a
+  /// [BoxDecoration] to the [decoration] parameter. This defaults to the
+  /// default gray iOS background color and 9 px corner radius.
+  ///
+  /// To set the text field's text padding, pass an [EdgeInsets] to the
+  /// [padding] parameter. This defaults to the default iOS padding.
+  ///
+  /// To set the icon colors for the search icon (prefix) and X-Mark (suffix),
+  /// pass a [Color] to the [itemColor] parameter. This defaults to the dark
+  /// gray iOS color.
+  ///
+  /// To set the padding insets for the search icon (prefix), pass an
+  /// [EdgeInsets] to the [prefixInsets] parameter. This defaults to the default
+  /// iOS insets.
+  ///
+  /// To set the padding insets for the X-Mark icon (suffix), pass an
+  /// [EdgeInsets] to the [suffixInsets] parameter. This defaults to the default
+  /// iOS insets.
+  ///
+  /// To dictate when the X-Mark (suffix) should be on, a.k.a. only on when
+  /// editing, not editing, on always, or on never, pass a
+  /// [OverlayVisibilityMode] to [suffixMode]. This defaults to only on when
+  /// editing.
+  ///
+  /// To customize the X-Mark (suffix) action, pass a [Function()] to [onSuffixTap].
+  /// This defaults to clearing the text.
+  ///
+  CupertinoSearchTextField({
+    TextEditingController? controller,
+    Function(String)? onChanged,
+    Function(String)? onSubmitted,
+    String placeholder = _kDefaultPlaceholder,
+    TextStyle placeholderStyle = _kDefaultPlaceholderStyle,
+    BoxDecoration? decoration = _kDefaultBoxDecoration,
+    Color backgroundColor = _kDefaultBackgroundColor,
+    BorderRadius borderRadius = _kDefaultBorderRadius,
+    EdgeInsets padding = _kDefaultContentPadding,
+    Color itemColor = _kDefaultItemColor,
+    EdgeInsets prefixInsets = _kDefaultPrefixInsets,
+    EdgeInsets suffixInsets = _kDefaultSuffixInsets,
+    OverlayVisibilityMode suffixMode = _kDefaultSuffixMode,
+    Function()? onSuffixTap,
```

Comment:
Consider using VoidCallback (again it doesn't actually do anything except self-describe a bit).